src/rust/aggregator/sqlx_queries/enumerated_volume/update.sql

```diff
@@ -0,0 +1,109 @@
+INSERT INTO aggregator.enumerated_volume (volume_as_base, volume_as_quote, address, module, struct, generic_asset_name, last_indexed_txn)
+WITH base_volumes AS (
+    SELECT
+        size_to_base_indivisible_subunits(markets.market_id, SUM("size")) AS volume_as_base,
+        base_account_address AS "address",
+        base_module_name AS module,
+        base_struct_name AS struct,
+        base_name_generic AS generic_asset_name
+    FROM fill_events, api.markets
+    WHERE emit_address = maker_address
+    AND markets.market_id = fill_events.market_id
+    AND fill_events.txn_version > COALESCE((
+        SELECT last_indexed_txn
+        FROM aggregator.enumerated_volume
+        WHERE (
+            markets.base_account_address = enumerated_volume."address"
+            AND markets.base_module_name = enumerated_volume.module
+            AND markets.base_struct_name = enumerated_volume.struct
+            AND enumerated_volume.generic_asset_name = ''
+        )
+        OR (
+            enumerated_volume."address" = ''
+            AND enumerated_volume.module = ''
+            AND enumerated_volume.struct = ''
+            AND enumerated_volume.generic_asset_name = markets.base_name_generic
+        )
+    ), 0)
+    GROUP BY
+        markets.market_id,
+        base_account_address,
+        base_module_name,
+        base_struct_name,
+        base_name_generic
+),
+quote_volumes AS (
+    SELECT
+        SUM(size_and_price_to_quote_indivisible_subunits(markets.market_id, "size", price)) AS volume_as_quote,
+        quote_account_address AS "address",
+        quote_module_name AS module,
+        quote_struct_name AS struct
+    FROM fill_events, api.markets
+    WHERE emit_address = maker_address
+    AND fill_events.txn_version > COALESCE((
+        SELECT last_indexed_txn
+        FROM aggregator.enumerated_volume
+        WHERE (
+            markets.base_account_address = enumerated_volume."address"
+            AND markets.base_module_name = enumerated_volume.module
+            AND markets.base_struct_name = enumerated_volume.struct
+            AND enumerated_volume.generic_asset_name = ''
+        )
+        OR (
+            enumerated_volume."address" = ''
+            AND enumerated_volume.module = ''
+            AND enumerated_volume.struct = ''
+            AND enumerated_volume.generic_asset_name = markets.base_name_generic
```

Comment:
Why is this selecting generic base name?

Since this is quote I am confused why this isn't just selecting `WHERE` there is a quote match